### PR TITLE
Add libebm, the native library for the conda-forge package interpret

### PR DESCRIPTION
**Pull request by paulbkoch**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by paulbkoch** [comment]
@conda-forge-admin, please rerender

**Comment by paulbkoch** [comment]
@conda-forge-admin, please rerender

**Comment by paulbkoch** [comment]
@conda-forge-admin, please rerender

**Comment by paulbkoch** [comment]
@conda-forge-admin, please rerender

**Comment by paulbkoch** [comment]
@conda-forge/help-c-cpp, ready for review!

**Comment by paulbkoch** [comment]
This package is needed to implement the partitioning mentioned by @carterbox in https://github.com/conda-forge/conda-forge.github.io/issues/1900

**Comment by carterbox** [comment]
What is the new package structure that this feedstock solves? i.e. Why isn't this just an output of interpret-core?

**Comment by paulbkoch** [comment]
The two benefits of having a new feedstock, which you pointed out in https://github.com/conda-forge/conda-forge.github.io/issues/1900#issuecomment-1434050358 @carterbox, were:
1) Reduceing build times and avoiding redundant builds
2) being able to have a conda_build_config.yaml and conda-forge.yml that would only apply to the native build

I would like to make the interpret-core and interpret packages both noarch, and have this native package compile for all the supported native architectures that are supported by conda-forge, which makes the conda-forge.yml problematic in a single unified feedstock.

I do think however that merging the interpret-core and interpret feedstocks afterwards might make sense, but I haven't come to that bridge yet.

**Comment by carterbox** [comment]
Fine, but also consider that there are 12 staged recipes ahead of you in the [c-cpp review queue](https://github.com/conda-forge/staged-recipes/pulls?q=is%3Apr+is%3Aopen+sort%3Acreated-asc+label%3Areview-requested+label%3Ac-cpp), you will get your package deployed faster if you prototype your recipes on your local machine and deploy to your exisiting feedstocks.

**Comment by paulbkoch** [comment]
Understood, and thank you for your help maintaining conda-forge.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/libebm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/interpretml/interpret/archive/200f9c3f08da6f17bdbeaee0e3285a2458f92f18.tar.gz
+  # curl -sL https://github.com/interpretml/interpret/archive/200f9c3f08da6f17bdbeaee0e3285a2458f92f18.tar.gz | openssl sha256
+  sha256: e61f014fb24f0dae58e4099f6affd15713928e66eab5078892e546b704920b8b
+  # url: https://github.com/interpretml/interpret/archive/v{{ version }}.tar.gz
```
[the remark this answers is not in the document]
Comment (paulbkoch) [reply recipes/libebm/meta.yaml]: The native library could be used separately and could also be updated independently from the python package in the future.  We haven't really advertised this yet though or supported it.  I don't mind matching the interpret version numbers for now though, so I have updated it to 0.3.1 to coincide with the next release of interpret, which I will put out as soon as we resolve these partitioning questions.

File: recipes/libebm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+build:
+  number: 0
```
[the remark this answers is not in the document]
Comment (paulbkoch) [reply recipes/libebm/meta.yaml:15]: Hi @carterbox -- I pushed an update that included this run_exports change along with version update number change.

**Comment by paulbkoch** [comment]
https://github.com/orgs/conda-forge/teams/help-c-cpp, ready for review!

**Comment by paulbkoch** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/libebm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+
+test:
+  commands: |
+    where libebm.dll  # [win]
+    # IF NOT EXIST "%LIBRARY_BIN%\libebm.dll" EXIT /B 1  # [win]
+    objdump -T "$PREFIX/lib/libebm.so"  # [linux]
+    # [ -e "$PREFIX/lib/libebm.dylib" ]  # [osx]
+    otool -L "$PREFIX/lib/libebm.dylib"  # [osx]
```
Comment (carterbox) [inline recipes/libebm/meta.yaml:33]:
```suggestion
  commands:
    - if not exist %LIBRARY_BIN%\libebm.dll exit 1  # [win]
    - test -f $PREFIX/lib/libebm.so       # [linux]
    - test -f $PREFIX/lib/libebm.dylib     # [osx]
```

**Pull request merged**